jme3-core/src/main/java/com/jme3/effect/ParticleEmitter.java

```diff
@@ -1117,6 +1117,68 @@ private void updateParticleState(float tpf) {
 
         vars.release();
     }
+    
+    private ColorRGBA startColorToFadeTo;
+    private ColorRGBA initialStartColorToFadeFrom;
+    private float totalStartColorFadeDuration;
+    private float startColorFadeDuration;
+    
+    private ColorRGBA endColorToFadeTo;
+    private ColorRGBA initialEndColorToFadeFrom;
+    private float totalEndColorFadeDuration;
+    private float endColorFadeDuration;    
+    
+    public void setFadeToStartColor(float fadeDuration, ColorRGBA newStartColor){
+        totalStartColorFadeDuration = fadeDuration;  
+        startColorFadeDuration = fadeDuration;
+        startColorToFadeTo = newStartColor;        
+        initialStartColorToFadeFrom = this.getStartColor().clone();
+    }
+    
+    public void setFadeToEndColor(float fadeDuration, ColorRGBA newEndColor){
+        totalEndColorFadeDuration = fadeDuration;
+        endColorFadeDuration = fadeDuration;
+        endColorToFadeTo = newEndColor;
+        initialEndColorToFadeFrom = this.getEndColor().clone();
+    }
+     
+     //convenience method for fading both start and end color over the same duration
+    public void setFadeToColors(float fadeDuration, ColorRGBA newStartColor, ColorRGBA newEndColor){
+        setFadeToEndColor(fadeDuration, newEndColor);
+        setFadeToStartColor(fadeDuration, newStartColor);
+    }  
+    private void updateColorFading(float tpf){
+        if(startColorFadeDuration > 0){
+            startColorFadeDuration -= tpf;
+            
+            if(startColorFadeDuration < 0){
+                setStartColor(startColorToFadeTo);
+            }else{
+                float timeSlerpPct = 1 - (startColorFadeDuration / totalStartColorFadeDuration);
+                
+                ColorRGBA slerpedStartColor = initialStartColorToFadeFrom.clone();
+                slerpedStartColor.interpolateLocal(startColorToFadeTo, timeSlerpPct);
+                setStartColor(slerpedStartColor);
+            }
+            
+        }
+        
+        //color fading for end color
+        if(endColorFadeDuration > 0){
+            endColorFadeDuration -= tpf;
+            if(endColorFadeDuration < 0){
+                setEndColor(endColorToFadeTo);
+            }else{
+                float timeSlerpPct =  1 - (endColorFadeDuration / totalEndColorFadeDuration);
+                
+                ColorRGBA slerpedEndColor = initialEndColorToFadeFrom.clone();
```

Comment:
The clone here creates a new object and it should be avoided. 
Also i am not sure if this is a good approach, this code replaces the end color that is chosen by the developer and this is an unexpected behaviour, this fading should probably be applied directly in the updateParticle method to every particle's color.